### PR TITLE
[Opt](exec) opt the dispose nullable column logic

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

1. remove some useless copy to referenc
2. erase block by batch
3. remove unless virtual function call

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/core/block.cpp [hunk 1]
```diff
@@ -177,15 +177,26 @@ void Block::erase(const std::set<size_t>& positions) {
...
 }
 
-void Block::erase(size_t position) {
-    if (data.empty()) {
-        LOG(FATAL) << "Block is empty";
+void Block::erase_tail(size_t start) {
+    DCHECK(start <= data.size()) << fmt::format(
+            "Position out of bound in Block::erase(), max position = {}", data.size());
+    data.erase(data.begin() + start, data.end());
+    for (auto it = index_by_name.begin(); it != index_by_name.end();) {
```
Comment (BiteTheDDDDt) [inline be/src/vec/core/block.cpp:184]: better change it to index_by_name.erase(key) for each data erased

Comment (HappenLee) [reply be/src/vec/core/block.cpp:184]: but also change the value of loc，so need for loop all key

Comment (HappenLee) [reply be/src/vec/core/block.cpp:184]: but also change the value of loc，so need for loop all key

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**